### PR TITLE
{AKS} Replace Workload Identity related functions in aks-preview with azure-cli

### PR DESCRIPTION
**Pull request by jiashun0011**
This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
```bash
az aks create --subscription $SUB_ID \
  -n $CLUSTER_NAME \
  -g $RG_NAME \
  --enable-workload-identity

az aks update --subscription $SUB_ID \
  -n $CLUSTER_NAME \
  -g $RG_NAME \
  --disable-workload-identity
```

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [x] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
AKS

**Comment by FumingZhang** [comment]
@jiashun0011, please fix failed CI check

File: src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py [hunk 1]
```diff
@@ -5924,7 +5924,7 @@ def test_update_workload_identity_profile__default_value_mc_enabled(self):
         )
         dec.context.attach_mc(mc)
         updated_mc = dec.update_workload_identity_profile(mc)
-        self.assertIsNone(updated_mc.security_profile.workload_identity)
+        self.assertIsNotNone(updated_mc.security_profile.workload_identity)
```
Comment (FumingZhang) [inline src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py:5927]: this seems to be a behavior change, can you explain it?

**Review by FumingZhang — COMMENTED** [review]
Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=84913249&view=results) to validate the change


> Replace Workload Identity related functions with stable version

Can you clarify in which version the code was added to official azure-cli and the version of azure-cli the current aks-preview depends on?

File: src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py (hunk 1 above)
Comment (jiashun0011) [reply src/aks-preview/azext_aks_preview/tests/latest/test_managed_cluster_decorator.py:5927]: > this seems to be a behavior change, can you explain it?

Currently `update_workload_identity_profile`'s behaviors differ between GA and Preview.

And this test fix meets stable cli's expectation.

**Comment by jiashun0011** [comment]
> Queued [live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=84913249&view=results) to validate the change
> 
> > Replace Workload Identity related functions with stable version
> 
> Can you clarify in which version the code was added to official azure-cli and the version of azure-cli the current aks-preview depends on?

It should be [Version 2.47.0](https://github.com/MicrosoftDocs/azure-docs-cli/blob/main/docs-ref-conceptual/release-notes-azure-cli.md#april-04-2023)

**Review by FumingZhang — APPROVED** [review]
LGTM

[live test](https://dev.azure.com/msazure/CloudNativeCompute/_build/results?buildId=84913249&view=logs&j=b162b355-d59d-5864-ce0f-0a70f12dd28b&t=a26fe913-e2be-5062-b08e-d15f1acc7ea2) passed

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**